apis_v1/static/js/PlanDialog.js

```diff
@@ -0,0 +1,267 @@
+  }
+
+  function updateTips (t) {
+    tips
+      .text(t)
+      .addClass('ui-state-highlight');
+    setTimeout(() => {
+      tips.removeClass('ui-state-highlight', 1500);
+    }, 500);
+  }
+
+  function checkLength (o, n, min, max) {
+    if (o.val().length > max || o.val().length < min) {
+      o.addClass('ui-state-error');
+      updateTips(`Length of ${n} must be between ${min} and ${max}.`);
+      return false;
+    } else {
+      return true;
+    }
+  }
+
+  function checkNumber (o, n) {
+    if (Number.isNaN(o.val())) {
+      o.addClass('ui-state-error');
+      updateTips(`${n} must be a number.`);
+      return false;
+    } else {
+      return true;
+    }
+  }
+
+  function checkRegexp (o, regexp, n) {
+    if (!(regexp.test(o.val()))) {
+      o.addClass('ui-state-error');
+      updateTips(n);
+      return false;
+    } else {
+      return true;
+    }
+  }
+
+  function dollarsToCents (cost) {
+    costString = cost + ""; // convert to a string
+    if (costString.length === 0) {
+      return '';
+    } else if (costString.indexOf('.') >= 0) {
+      const money = (costString).split(".");
+      dollars = money[0];
+      cents = money[1];
+
+      if (cents.length > 2) {
+        return dollars + cents.substring(0, 2);
+      } else if (cents.length === 0) {
+        return dollars + '00';
+      } else if (cents.length === 1) {
+        return dollars + cents + '0'
+      } else if (cents.length === 2) {
+        return dollars + cents;
+      }
+    }
+    return cost + '00';
+  }
+
+  function saveNewPlan () {
+    let valid = true;
+    allFields.removeClass('ui-state-error');
+    expires = "";
+
+    valid = valid && checkLength(couponCode, 'Coupon Code', 3, 128);
+    valid = valid && checkNumber(listPriceMonthlyCredit, 'List Price Monthly');
+    valid = valid && checkNumber(discountedPriceMonthlyCredit, 'Discounted Price Monthly');
+    if (couponExpiresDate.val().length) {
+      valid = valid && checkRegexp(couponExpiresDate, /^([0-9]{2}\/[0-9]{2}\/[0-9]{4})+$/, 'Expiration date must use this format: 12/25/2022');
+      if (valid) {
+        moment().tz('America/Los_Angeles').format();
+        expires = moment(couponExpiresDate.val(), 'MM/DD/YYYY HH:mm').format();
+      }
+    }
+
+    if (valid) {
+      const newPlanData = {
+        couponCode: couponCode.val(),
+        planTypeEnum: planTypeEnum.val(),
+        couponAppliedMessage: couponAppliedMessage.val(),
+        hiddenPlanComment: hiddenPlanComment.val(),
+        listPriceMonthlyCredit: dollarsToCents(listPriceMonthlyCredit.val()),
+        discountedPriceMonthlyCredit: dollarsToCents(discountedPriceMonthlyCredit.val()),
+        featuresProvidedBitmap: featuresProvidedBitmap.val(),
+        couponExpiresDate: expires,
+      }
+
+      const apiURL = `${window.location.origin}/apis/v1/createNewPlan`;
+      $.getJSON(apiURL, newPlanData, () => { dialog.dialog('close'); location.reload(); })
```

Comment:
Does this (and `deleteThisPlan` below) API call include the voter_device_id cookie value so we can verify that the person doing the action has sufficient authority?

Reply:
Yes, all Dispatcher.loadEndpoint() calls include the device_id cookie